src/ci/docker/x86_64-gnu-tools/checktools.sh

```diff
@@ -57,17 +57,35 @@ verify_status() {
     fi
 }
 
+check_dispatch() {
+    if [ "$1" = submodule_changed ]; then
+        # ignore $2 (branch id)
+        verify_status $3 $4
+    else
+        if [ "$2" = beta ]; then
+            if grep -vq '"'"$1"'":"(test|build)-fail"' "$TOOLSTATE_FILE"; then
```

Comment:
1. You should just use `-q` here. The `-v` flag means inverted search 
2. `grep` by default uses BRE which means you need to escape the brackets and vertical bars:

    ```sh
    if grep -q '"'"$1"'":"\(test\|build\)-fail"' "$TOOLSTATE_FILE"; then
    ```

    or use `-E` to use ERE.

Reply:
done